### PR TITLE
Issue 94: Enable full statistic functionality

### LINKED ISSUE #94: Support different statistics for same variable
**Is your feature request related to a problem? Please describe.**
Currently we _de facto_ only support one statistic per variable in get_timeseries_data. USGS plugin is hardcoded to Mean (statCd = 00003).

**Describe the solution you'd like**
Enable multiple statistics to be returned for the same variable.
Enable querying by more than one statistic at the same time. I.e., return data that have Mean or Max
If/How this is handled in the column names tbd.

**Additional context**
BASIN-3D currently supports MIN, MAX, MEAN, TOTAL as part of the [MeasurementMetadataMixin](BASIN-3D/basin3d/blob/main/basin3d/core/models.py)

### PR DESCRIPTION
**Pull request by dschristianson**
Closes #94

- Adds querying functionality for statistic, i.e., single or multiple statistics can be requested from data sources
- Enable specified statistics to be returned from USGS Daily Values, i.e., removed hardcoded MEAN from USGS plugin
- Adds/updates unit and integration tests
- Fixes flake8 issues in all modules modified for this functionality

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update -- DONE.

## How Has This Been Tested?
- [x] Unit tests
- [x] Integration tests 
~- [ ] Test coverage >= 90%~
- [x] Flake8 Tests: a few known issues remain
- [x] Mypy Tests 
~- [ ] Other - <describe manual tests for the reviewer>~

### Test Configuration
* Python Version: 3.7.11

## PR Self Evaluation
- [x] My code follows the agreed upon best practices
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests or modified existing tests that prove my fix is effective or that my feature works
- [x] Existing unit tests pass locally with my changes
~- [ ] Any dependent changes have been merged and published in the appropriate modules~
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Comment by dschristianson** [comment]
@vchendrix: this PR is essentially ready for review. I'm waiting until we tag and version existing develop branch (see PR#103 before removing draft mode on this PR.

**Comment by vchendrix** [comment]
> @vchendrix: this PR is essentially ready for review. I'm waiting until we tag and version existing develop branch (see PR#103) before removing draft mode on this PR.

@dschristianson these look good. I added review for PR #103.  I have a question there.

File: basin3d/plugins/usgs.py [hunk 1]
```diff
@@ -636,8 +645,8 @@ def list(self, **kwargs):
                     # TODO: write some tests for this which will require mocking a data return.
                     # Only filter if quality_checked is True
                     # if QUERY_PARAM_RESULT_QUALITY not in kwargs or not kwargs[QUERY_PARAM_RESULT_QUALITY] or \
-                            # (QUERY_PARAM_RESULT_QUALITY in kwargs and kwargs[
-                                # QUERY_PARAM_RESULT_QUALITY] == result_quality):
+                    #         (QUERY_PARAM_RESULT_QUALITY in kwargs and kwargs[
+                    #          QUERY_PARAM_RESULT_QUALITY] == result_quality):
```
Comment (vchendrix) [inline basin3d/plugins/usgs.py:649]: Let's remove this commented out code and write an issue ticket for it.

**Review by vchendrix — CHANGES_REQUESTED** [review]
Looks great.  I just have one small suggestion.

File: basin3d/plugins/usgs.py (hunk 1 above)
Comment (dschristianson) [reply basin3d/plugins/usgs.py:649]: Any chance we can leave it for now? The next task is reworking result_quality so I would deal with this comment in the next PR. Would be easier for me to think about / address then. I'm not totally sure what needs to happen here.

Comment (vchendrix) [reply basin3d/plugins/usgs.py:649]: ok.

**Review by vchendrix — APPROVED** [review]

**Pull request merged**